File: src/structural/decorator/typescript/classes/TShirt.ts
import { ProductProtocol } from './interfaces/ProductProtocol';

export class TShirt implements ProductProtocol {
  private _price = 49.9;
  private _name = 'TShirt';

  get price(): number {
    return this._price;
  }
  get name(): string {
    return this._name;
  }
}

File: src/structural/decorator/python/classes/tshirt.py
from classes.interfaces.product_protocol import ProductProtocol


class TShirt(ProductProtocol):
    def __init__(self):
        self.price = 49.9
        self.name = 'TShirt'

    def get_price(self) -> int:
        return self.price

    def get_name(self) -> str:
        return self.name
